src/NuGet.Core/NuGet.Protocol.Core.v3/HttpSource/HttpSource.cs

```diff
@@ -568,6 +500,39 @@ public void Dispose()
             _httpClientLock.Dispose();
         }
 
+        private class ThrottledResponse : IDisposable
+        {
+            private IThrottle _throttle;
+            private readonly object _throttleLock = new object();
+
+            public ThrottledResponse(IThrottle throttle, HttpResponseMessage response)
+            {
+                if (response == null)
+                {
+                    throw new ArgumentNullException(nameof(response));
+                }
+
+                _throttle = throttle;
+                Response = response;
+            }
+            
+            public HttpResponseMessage Response { get; }
+
+            public void Dispose()
+            {
+                Response.Dispose();
+
+                lock (_throttleLock)
+                {
+                    if (_throttle != null)
```

Comment:
From my understanding you create the wrapper object regardless of the initial value of `_throttle`. So I related this null check to the case of initially null throttle only. I missed the double dispose scenario. My point was not to create this wrapper object when not throttle is provided. Though having `NullThrottle` would help here as well.